src/status_im/contexts/wallet/common/utils/wallet_connect.cljs

```diff
@@ -0,0 +1,21 @@
+(ns status-im.contexts.wallet.common.utils.wallet-connect
+  (:require ["@walletconnect/core" :refer [Core]]
+            ["@walletconnect/web3wallet" :refer [Web3Wallet]]
+            [promesa.core :as promesa]
+            [status-im.config :as config]))
+
+(defn- wallet-connect-core
+  []
+  (Core. (clj->js {:project-id config/wallet-connect-project-id})))
+
+(defn init
+  []
+  (let [core     (wallet-connect-core)
+        metadata (clj->js config/default-wallet-connect-metadata)]
+    (promesa/create
```

Comment:
@clauxx - looks good here?

Reply:
looks weird. If Web3Wallet.init already returns a promise, there's no need to create another promise with `promesa/create`. Just return the promise and it would be handled in the effect it's called from.

Reply:
@clauxx makes sense, I'll modify it 👍